src/info.c

```diff
@@ -1273,6 +1273,43 @@ static void info_shiplogView( unsigned int wid, char *str )
       dialogue_msgRaw( _("Log message"), tmp);
 }
 
+/**
+ * @brief Asks the player for an entry to add to the log
+ *
+ * @param wid Window widget
+ * @param str Button widget name
+ */
+static void info_shiplogAdd( unsigned int wid, char *str )
+{
+   char *tmp;
+   char *logname;
+   int logid;
+   (void) str;
+
+   logname = toolkit_getList( wid, "lstLogs" );
+   if ( logname == NULL || !strcmp( "All", logname ) ){
```

Comment:
Need parentheses around the individual statements and space between the curly brace and preceding parenthesis. Also, this isn't the right way to use strcmp; strcmp isn't a binary operation, it either returns -1, 0, or 1 depending on how it compares. So `( strcmp("All", logname) == 0 )` is the correct form.